Bugfix/postgres service

carrino opened the pull request.

cooper87 commented:
hi, carrino
    Sorry for submitting the question here,I could not find your e-mail,or a better way to connect you.I'm very interested in atlasdb,and I begin to to learn it by wiki or code.While I meet a very serious problem that KV API seems didn't work.
I take PostgreSQL as kv store,and startup atlasdb by 'com.palantir.atlasdb.endpoint.PostgresEndpoint server postgres_endpoint1.yml'.the console of eclipse show "started sucessfuly".then I post 'http://localhost:3818/keyvalue/get-all-table-names' or 'http://localhost:3818/keyvalue/initialize',the console print:
----------------------------------------------------------------------------------------------
ERROR [2016-01-18 02:36:23,466] io.dropwizard.jersey.errors.LoggingExceptionMapper: Error handling a request: b4ef5bd3d56617cd
! java.lang.NullPointerException: null
! at java.lang.String.<init>(String.java:556) ~[na:1.7.0_65]
! at com.palantir.atlasdb.keyvalue.partition.map.DynamicPartitionMapImpl.fromTable(DynamicPartitionMapImpl.java:1172) ~[bin/:na]
! at com.palantir.atlasdb.keyvalue.partition.map.InKvsPartitionMapService.getMap(InKvsPartitionMapService.java:114) ~[bin/:na]
! at com.palantir.atlasdb.keyvalue.partition.map.InKvsPartitionMapService.getMapVersion(InKvsPartitionMapService.java:119) ~[bin/:na]
! at com.palantir.atlasdb.endpoint.PostgresEndpoint$2.get(PostgresEndpoint.java:94) ~[bin/:na]
----------------------------------------------------------------------------------------------
I am in the trouble serval days and try to find the reason, but I am not luck.Please give me a hand to resolve this proleam or show me anoter a way to use kv api.
Thanks in advance ^ ^.

carrino commented:
This is a good question and a fine place to post it.  The Partition stuff
isn't 100% ready to go with postgres.

If you want to just run one node with postgres, I'd just use postgres
standalone.

If you want to run a cluster, I'd recommend using cassandra for that.

We have also changed the way you configure postgresdb and I updated the
sample config here: https://github.com/palantir/atlasdb/pull/168

With the new config format i think this should work: running with working
dir src/dist/ and execute "AtlasSever
var/conf/atlasdb_standalone_postgres.yml"


On Sun, Jan 17, 2016 at 10:05 PM, cooper87 <notifications@github.com> wrote:

> hi, carrino
> Sorry for submitting the question here,I could not find your e-mail,or a
> better way to connect you.I'm very interested in atlasdb,and I begin to to
> learn it by wiki or code.While I meet a very serious problem that KV API
> seems didn't work.
> I take PostgreSQL as kv store,and startup atlasdb by
> 'com.palantir.atlasdb.endpoint.PostgresEndpoint server
> postgres_endpoint1.yml'.the console of eclipse show "started
> sucessfuly".then I post '
> http://localhost:3818/keyvalue/get-all-table-names' or '
> http://localhost:3818/keyvalue/initialize',the console print:
>
> ERROR [2016-01-18 02:36:23,466]
> io.dropwizard.jersey.errors.LoggingExceptionMapper: Error handling a
> request: b4ef5bd3d56617cd
> ! java.lang.NullPointerException: null
> ! at java.lang.String.(String.java:556) ~[na:1.7.0_65]
> ! at
> com.palantir.atlasdb.keyvalue.partition.map.DynamicPartitionMapImpl.fromTable(DynamicPartitionMapImpl.java:1172)
> ~[bin/:na]
> ! at
> com.palantir.atlasdb.keyvalue.partition.map.InKvsPartitionMapService.getMap(InKvsPartitionMapService.java:114)
> ~[bin/:na]
> ! at
> com.palantir.atlasdb.keyvalue.partition.map.InKvsPartitionMapService.getMapVersion(InKvsPartitionMapService.java:119)
> ~[bin/:na]
> ! at
> com.palantir.atlasdb.endpoint.PostgresEndpoint$2.get(PostgresEndpoint.java:94)
> ~[bin/:na]
>
> I am in the trouble serval days and try to find the reason, but I am not
> luck.Please give me a hand to resolve this proleam or show me anoter a way
> to use kv api.
> Thanks in advance ^ ^.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/palantir/atlasdb/pull/161#issuecomment-172433787>.
>

cooper87 commented:
Thank you very much,I've tried again as your said,and succeeded.I will try cassandra next step.have a good day ^ ^.
